src/Doctrine/EntityClassGenerator.php

```diff
@@ -69,6 +71,15 @@ public function generateRepositoryClass(string $repositoryClass, string $entityC
     {
         $shortEntityClass = Str::getShortClassName($entityClass);
         $entityAlias = strtolower($shortEntityClass[0]);
+
+        $passwordUserInterfaceName = UserInterface::class;
+
+        if (interface_exists(PasswordAuthenticatedUserInterface::class)) {
+            $passwordUserInterfaceName = PasswordAuthenticatedUserInterface::class;
+        }
+
+        $interfaceClassNameDetails = new ClassNameDetails($passwordUserInterfaceName, 'Symfony\Component\Security\Core\User');
```

Comment:
Why do we need `ClassNameDetails` here? Can't we just use the normal class strings?

You might be doing this due to the logic in the template to figure out *which* location the `use` statement should be. I'd prefer passing in a second flag to know that - e.g. `is_using_password_authenticated_interface`. That should simplify the template

Reply:
We need the full class name for the `use` statement and the shortName for the type hint in `upgradePassword()`.

 We currently already have a `$with_password_upgrade` var available in the templates - we could add a `$is_using_password_authenticated_interface` var as well, but i think that would make things even messier in the template when we need to determine the sort order of the use statements.  Ultimately what we really need is a use statement sorter to solve the messy part - but I'm open to ideas.

Also to consider, #824  introduces the concept of a `TemplateClassDetails` object the returns formatted strings for the particular use case. e.g.

```php
$userDetails = new TemplateClassDetails(User::class, true);    // second arg === $isTyped

$userDetails->getUseStatement();
// use App\Entity\User;

$userDetails->getConstructorArg();
// User $user

$userDetails->getMethodArg();  // similar to the above but spacing is slightly different
// User $user

$userDetails->getVariable();
// $user

$userDetails->getProperty();
// $this->user

// All of the above add appropriate spacing, casing, trailing commas, new lines, etc..
```
While 824 is not ready yet, ideally _if_ we merge that in with, a lot of stuff in this PR will be refactored to leverage the new helper and cleanup the templates.

Reply:
Ok then!

> Ultimately what we really need is a use statement sorter to solve the messy part

👍